src/main/java/com/couchbase/lite/replicator/BulkDownloader.java

```diff
@@ -120,29 +121,37 @@ protected void executeRequest(HttpClient httpClient, HttpUriRequest request) {
             }
             StatusLine status = response.getStatusLine();
             if (status.getStatusCode() >= 300) {
-                // conflict could be happen. So ERROR prints make developer confused.
-                if (status.getStatusCode() == 409)
-                    Log.w(Log.TAG_REMOTE_REQUEST, "Got error status: %d for %s.  Reason: %s",
-                            status.getStatusCode(), request, status.getReasonPhrase());
-                else
-                    Log.e(Log.TAG_REMOTE_REQUEST, "Got error status: %d for %s.  Reason: %s",
-                            status.getStatusCode(), request, status.getReasonPhrase());
-                error = new HttpResponseException(status.getStatusCode(),
-                        status.getReasonPhrase());
+                try {
+                    // conflict could be happen. So ERROR prints make developer confused.
+                    if (status.getStatusCode() == 409)
+                        Log.w(Log.TAG_REMOTE_REQUEST, "Got error status: %d for %s.  Reason: %s",
+                                status.getStatusCode(), request, status.getReasonPhrase());
+                    else
+                        Log.e(Log.TAG_REMOTE_REQUEST, "Got error status: %d for %s.  Reason: %s",
+                                status.getStatusCode(), request, status.getReasonPhrase());
+                    error = new HttpResponseException(status.getStatusCode(),
+                            status.getReasonPhrase());
```

Comment:
There is a possibility that respondWithResult will be called twice on line#175 and 206.